pyccel/ast/numpyext.py

```diff
@@ -786,22 +786,22 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
+    _shape = ()
+    _rank  = 0
 
     def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
-            sh = list(sh)
-            del sh[self.dim]
-            self._shape = tuple(sh)
-        else:
-            self._shape = ()
-        self._rank = len(self._shape)
```

Comment:
Why did you remove this? The shape is now constant. Does this mean we have lost the implementation of the optional `axis` argument? If there is no code associated with the `dim` parameter then I don't think it's a problem I was just wondering if you thought about this?